docs/Antora/modules/howto_guides/pages/howto_configure_instance_providers.adoc

```diff
@@ -0,0 +1,206 @@
+= Instance Configuration with Providers
+
+A Config Provider is an interface that enriches the input options dictionary with values from external sources such as a JSON file, environment variables, or command line arguments. A Config Provider can be assigned to the instance builder through the method `addConfigProvider`. This method does not override the options dictionary; instead, it retains the provider in the builder to augment options while building the instance.
+
+**Related articles:**
+
+- xref:howto_configure_instance.adoc#instance[Architecture - Instance]
+
+== Instance Configuration with JSON Provider
+
+An instance can be configured from a JSON file. An example of a JSON file is listed below. JSON keys are not case-sensitive, so the keys "MODULEMANAGER", "modulemanager", and "MoDuLeMaNaGeR" are recognized as the same key. If a value has the wrong type, or there are additional key-value pairs, they will be ignored. If a provider is overriding a list value, the old list items will be erased.
```

Comment:
The "MoDuLeMaNaGeR" example is a bit extra :D I think just stating that it's case-insensitive is enough.